libctru/source/system/syscalls.c

```diff
@@ -168,7 +164,7 @@ void initThreadVars(struct Thread_tag *thread)
 	tv->thread_ptr = thread;
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Warray-bounds"
-	tv->tls_tp = (thread != NULL ? (u8*)thread->stacktop : __tls_start) - 8; // Arm ELF TLS ABI mandates an 8-byte header
+	tv->tls_tp = (thread != NULL ? (u8*)thread->stacktop : __tls_start);
```

Comment:
This is technically a change in behavior. 8 additional bytes are allocated for each thread to be used as the header, which seemed necessary for the alignment to be correct as I was experimenting, but there might be a way to avoid it.

Reply:
The double pointer sized TLS header is only used by more general, shared-library-oriented TLS models that we don't use (we effectively use `local-exec`), thus the reason why libctru (and libnx) have always faked the existence of the header by simply subtracting its size from the real start of the TLS area. I am not sure why you have had problems with alignment, I can only think of the fact that the *stack* on 32-bit machines typically needs 8-byte alignment, so that should be the baseline.

Reply:
It took a bit of trial and error to get right, since the location of the header also needs to be aligned properly, but I think I figured it out, and pushed my changes.

This appears to work with all the test cases I listed above in the description as well.